src/main/java/arb/logic/commands/project/MarkProjectCommand.java

```diff
@@ -0,0 +1,57 @@
+package arb.logic.commands.project;
+
+import static arb.logic.commands.client.DeleteClientCommand.MESSAGE_DELETE_CLIENT_SUCCESS;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import arb.commons.core.Messages;
+import arb.commons.core.index.Index;
+import arb.logic.commands.Command;
+import arb.logic.commands.CommandResult;
+import arb.logic.commands.exceptions.CommandException;
+import arb.model.ListType;
+import arb.model.Model;
+import arb.model.project.Project;
+
+/**
+ * Marks a project identified using it's displayed index from the address book.
+ */
+public class MarkProjectCommand extends Command {
+
+    public static final String COMMAND_WORD = "mark";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Marks the project identified by the index number used in the displayed project list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_MARK_PROJECT_SUCCESS = "Marked Project: %1$s";
+
+    private final Index targetIndex;
+
+    public MarkProjectCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model, ListType currentListBeingShown) throws CommandException {
+        requireNonNull(model);
+        List<Project> lastShownList = model.getFilteredProjectList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PROJECT_DISPLAYED_INDEX);
+        }
+        Project projectToMark = lastShownList.get(targetIndex.getZeroBased());
+        model.markProject(projectToMark);
+
```

Comment:
Can you add
```
model.setProject(projectToMark, projectToMark);
model.updatedFilteredProjectList(PREDICATE_SHOW_ALL_PROJECTS);
```
here? And see if it works? If it throws an exception then we'll probably have to make changing Status produce a new Project object.

Reply:
It works holy fishcakes